spi/common/jwt-signer-spi/src/main/java/org/eclipse/edc/jwt/signer/spi/JwsSignerProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.jwt.signer.spi;
+
+import com.nimbusds.jose.JWSSigner;
+
+import java.util.function.Function;
+
+/**
+ * A JwsSignerProvider provides a {@link JWSSigner} for a given private key ID.
+ */
+@ExtensionPoint
+public interface JwsSignerProvider extends Function<String, JWSSigner> {
```

Comment:
I would define an explicit method instead of extending the java `Function`, pretty much because that will make easier to find usages in the IDE (plus it will be semantically better)